### ISSUE TITLE
Provide a file watcher thing

### ISSUE THREAD

**Issue by steelbrain** [steelbrain/Universal-Compiler#4]

**Comment by steelbrain** [comment]
Not happening any time soon when I look at my TODO list.

**Update**: I have specified the Default Options for multiple FileTypes and have rewritten the WatcherH::ScanDir function.
So all we need now is to enhance our Manifest File and Convert from `Files`, `Folders` and `Exluded` to `Entries` and `Tree`.
The Tree will be a nested thing that'll help us set-up the UI.
The Entries are gonna be Objects with Full Paths as keys and `IncludedFiles`, `Ext`, `FileType`, `FileName` and `IsCompressed` as options.

The only thing I am not sure about right now is whether we should keep the `Tree` thing in Universal-Compiler or separate it into DeProc, 'cause we're only gonna use it there. I mean not **everyone** who uses Universal-Compiler is gonna use it in DeProc.

**Update**: I chose to keep the Tree entries in Universal-Compiler, Because it's a component of DeProc and because The JSON file it creates is called `DeProc.json`. We have a working file watcher as of now, But it's not awesome, I mean the Runtime options aren't persisted to Manifest,

I am thinking of creating a modular system for it, just like the one we have in the Compiler Section, Each Plugin will control the watching, Execution and Manifest Updates for it's extension(s).
